### PR TITLE
Update docs to reference how projects are handled

### PR DESCRIPTION
**Pull request by loosebazooka**
standard -> appengine-web.xml based
flexible -> app.yaml based (now include app.yaml based standard projects)

### CODE REVIEW

File: USER_GUIDE.md [hunk 1]
```diff
@@ -17,24 +17,26 @@ buildscript {
   }
 }
 
-apply plugin: "com.google.cloud.tools.appengine-standard"
+apply plaugin: "com.google.cloud.tool.appengine" // for automatic environment determination
```
Comment (TadCordle) [inline USER_GUIDE.md]: ~~plaugin~~ plugin

File: USER_GUIDE.md [hunk 2]
```diff
@@ -17,24 +17,26 @@ buildscript {
...
 // or
-apply plugin: "com.google.cloud.tools.appengine-flexible"
+apply plugin: "com.google.cloud.tools.appengine-appenginewebxml"
+// or
+apply plugin: "com.google.cloud.tools.appengine-appyaml"
 ```
 
-You can also use the `com.google.cloud.tools.appengine` plugin that will automatically determine
+When you use the `com.google.cloud.tools.appengine` plugin that will automatically determine
```
Comment (TadCordle) [inline USER_GUIDE.md]: This sentence is a bit confusing.

File: USER_GUIDE.md [hunk 3]
```diff
@@ -207,9 +209,9 @@ Use the following tasks :
 The deployment source directory can be overridden by setting the `appEngineDirectory` parameter
 in the deploy configuration.
 
-For standard it defaults to `${buildDir}/staged-app/WEB-INF/appengine-generated`, you should probably
-not change this configuration, for standard configured projects, this is the location that your
-xml configs are converted into yaml for deployment.
+For appengine-web.xml based projects it defaults to `${buildDir}/staged-app/WEB-INF/appengine-generated`,
```
Comment (TadCordle) [inline USER_GUIDE.md]: Some punctuation/rewording, maybe something like:

>For appengine-web.xml based projects, it defaults to `...`. You should not change this configuration; this is the location where your xml configs are converted into yaml for deployment."

File: USER_GUIDE.md (hunk 2 above)
Comment (loosebazooka) [reply USER_GUIDE.md]: reworded it a little.

File: USER_GUIDE.md [hunk 4]
```diff
@@ -17,24 +17,26 @@ buildscript {
   }
 }
 
-apply plugin: "com.google.cloud.tools.appengine-standard"
+apply plugin: "com.google.cloud.tool.appengine" // for automatic environment determination
```
Comment (TadCordle) [inline USER_GUIDE.md]: `tool` -> `tools`?

File: src/main/resources/META-INF/gradle-plugins/com.google.cloud.tools.appengine-appenginewebxml.properties [hunk 5]
```diff
@@ -0,0 +1,16 @@
+#
```
Comment (TadCordle) [inline src/main/resources/META-INF/gradle-plugins/com.google.cloud.tools.appengine-appenginewebxml.properties]: What's this file for?

Comment (loosebazooka) [reply src/main/resources/META-INF/gradle-plugins/com.google.cloud.tools.appengine-appenginewebxml.properties]: it just allows the user to use "appengine-appenginewebxml" as a plugin
```
apply plugin: 'appengine-appenginewebxml'
```

**Pull request merged**